ocp_resources/resource.py

```diff
@@ -107,13 +108,26 @@ def get_client(
             )
         )
     try:
-        # Ref: https://github.com/kubernetes-client/python/blob/v26.1.0/kubernetes/base/config/__init__.py
-        LOGGER.info("Trying to get client via new_client_from_config")
-
         # kubernetes.config.kube_config.load_kube_config sets KUBE_CONFIG_DEFAULT_LOCATION during module import.
         # If `KUBECONFIG` environment variable is set via code, the `KUBE_CONFIG_DEFAULT_LOCATION` will be None since
         # is populated during import which comes before setting the variable in code.
         config_file = config_file or os.environ.get("KUBECONFIG", "~/.kube/config")
+        proxy = os.environ.get("HTTPS_PROXY") or os.environ.get("HTTP_PROXY")
```

Comment:
`proxy` should be passed as arg

Reply:
Why?
Using those environment variables are a common way to configure proxy in Python...

Reply:
I don't think this is a common way in Python.

Maybe a common way to define PROXY to applications in general, anyway since oc/kubeconfig use environment variables to get the proxy we can use it here, but you need to add it as arg as well

Reply:
Got it, done.
Please note that the additional argument will not take into affect for internal calls of get_client, like in
```
    def _set_client_and_api_version(self) -> None:
        if not self.client:
            self.client = get_client(config_file=self.config_file, context=self.context)

        if not self.api_version:
            self.api_version = _get_api_version(dyn_client=self.client, api_group=self.api_group, kind=self.kind)
```
Which is being called internally by 'exists' property.